File: FLASH-FLOOD/backend/app/schemas/seismic.py
from pydantic import BaseModel
from .common import DemoMetadata


class SeismicResponse(DemoMetadata):
    location_id: str
    event_id: str
    magnitude: float
    depth_km: int
    distance_km: int
    event_time: str
    latitude: float
    longitude: float
    seismic_status: str
    note: str
    earthquake_prediction_provided: bool = False
    secondary_hazards: list[dict[str, str]]

File: FLASH-FLOOD/frontend/src/types/seismic.ts
export interface SecondaryHazard { name: string; status: string }

export type SeismicStatus =
  | 'MONITORING'
  | 'NO RECENT EVENT'
  | 'DATA UNAVAILABLE'
  | 'DEMO ALERT'

export interface SeismicEvent {
  id: string | null
  magnitude: number | null
  depthKm: number | null
  distanceKm: number | null
  time: string
  latitude: number | null
  longitude: number | null
  status: SeismicStatus
  note: string
  dataStatus?: string
  secondaryHazards?: SecondaryHazard[]
}